diff --git a/clang/cmake/caches/Fuchsia-stage2.cmake b/clang/cmake/caches/Fuchsia-stage2.cmake
index de14bde2007a..7dce396d076d 100644
--- a/clang/cmake/caches/Fuchsia-stage2.cmake
+++ b/clang/cmake/caches/Fuchsia-stage2.cmake
@@ -2,7 +2,7 @@
 
 option(FUCHSIA_ENABLE_LLDB "Enable LLDB")
 
-set(LLVM_TARGETS_TO_BUILD X86;ARM;AArch64;RISCV CACHE STRING "")
+set(LLVM_TARGETS_TO_BUILD X86;ARM;AArch64;RISCV;LoongArch CACHE STRING "")
 
 set(PACKAGE_VENDOR Fuchsia CACHE STRING "")
 
@@ -138,7 +138,7 @@ if(WIN32 OR LLVM_WINSYSROOT)
   set(RUNTIMES_${target}_CMAKE_MODULE_LINKER_FLAGS ${WINDOWS_LINK_FLAGS} CACHE STRING "")
 endif()
 
-foreach(target aarch64-unknown-linux-gnu;armv7-unknown-linux-gnueabihf;i386-unknown-linux-gnu;riscv64-unknown-linux-gnu;x86_64-unknown-linux-gnu)
+foreach(target aarch64-unknown-linux-gnu;armv7-unknown-linux-gnueabihf;i386-unknown-linux-gnu;loongarch64-unknown-linux-gnu;riscv64-unknown-linux-gnu;x86_64-unknown-linux-gnu)
   if(LINUX_${target}_SYSROOT)
     # Set the per-target builtins options.
     list(APPEND BUILTIN_TARGETS "${target}")
@@ -449,7 +449,7 @@ set(LLVM_RUNTIME_TARGETS "${RUNTIME_TARGETS}" CACHE STRING "")
 
 # Setup toolchain.
 set(LLVM_INSTALL_TOOLCHAIN_ONLY ON CACHE BOOL "")
-set(LLVM_TOOLCHAIN_TOOLS
+set(TOOLCHAIN_TOOLS
   dsymutil
   llvm-ar
   llvm-cov
@@ -466,7 +466,6 @@ set(LLVM_TOOLCHAIN_TOOLS
   llvm-libtool-darwin
   llvm-lipo
   llvm-ml
-  llvm-mt
   llvm-nm
   llvm-objcopy
   llvm-objdump
@@ -485,8 +484,13 @@ set(LLVM_TOOLCHAIN_TOOLS
   llvm-xray
   opt-viewer
   sancov
-  scan-build-py
-  CACHE STRING "")
+  scan-build-py)
+
+if(LLVM_ENABLE_LIBXML2)
+  list(APPEND TOOLCHAIN_TOOLS llvm-mt)
+endif()
+
+set(LLVM_TOOLCHAIN_TOOLS "${TOOLCHAIN_TOOLS}" CACHE STRING "")
 
 set(LLVM_Toolchain_DISTRIBUTION_COMPONENTS
   bolt

diff --git a/clang/cmake/caches/Fuchsia.cmake b/clang/cmake/caches/Fuchsia.cmake
index 83ff4ccc167d..66a775f9abad 100644
--- a/clang/cmake/caches/Fuchsia.cmake
+++ b/clang/cmake/caches/Fuchsia.cmake
@@ -2,7 +2,7 @@
 
 option(FUCHSIA_ENABLE_LLDB "Enable LLDB")
 
-set(LLVM_TARGETS_TO_BUILD X86;ARM;AArch64;RISCV CACHE STRING "")
+set(LLVM_TARGETS_TO_BUILD X86;ARM;AArch64;LoongArch;RISCV CACHE STRING "")
 
 set(PACKAGE_VENDOR Fuchsia CACHE STRING "")
 
